### PR TITLE
fix: web control plane endpoint and auth

### PR DESCRIPTION
**Pull request by bruuuuuuuce**
- fixes build so that build gets run on each pr
- fixes control plane client to use correct endpoint

Integration tests now pass in my dev cell using the updated envoy config from https://github.com/momentohq/control-plane-service/pull/306

### CODE REVIEW

File: packages/client-sdk-web/src/internal/control-client.ts [hunk 1]
```diff
@@ -76,10 +74,12 @@ export class ControlClient<
     //   configuration: props.configuration,
     // });
     console.log(
-      `\n\n\nCreating control client with endpoint: ${props.credentialProvider.getCacheEndpoint()}\n\n\n`
+      `\n\n\nCreating control client with endpoint: ${props.credentialProvider.getControlEndpoint()}\n\n\n`
```
Comment (bruuuuuuuce) [inline packages/client-sdk-web/src/internal/control-client.ts:77]: 😄

File: packages/client-sdk-web/src/internal/control-client.ts [hunk 2]
```diff
@@ -76,10 +74,12 @@ export class ControlClient<
...
     );
+
+    this.authToken = props.credentialProvider.getAuthToken();
     this.clientWrapper = new control.ScsControlClient(
-      `https://${props.credentialProvider.getCacheEndpoint()}`,
+      `https://${props.credentialProvider.getControlEndpoint()}`,
```
Comment (bruuuuuuuce) [inline packages/client-sdk-web/src/internal/control-client.ts:82]: 😄

File: packages/client-sdk-web/src/internal/control-client.ts [hunk 3]
```diff
@@ -130,7 +130,7 @@ export class ControlClient<
     return await new Promise<DeleteCache.Response>(resolve => {
       this.clientWrapper.deleteCache(
         request,
-        null,
+        {authorization: this.authToken},
```
Comment (bruuuuuuuce) [inline packages/client-sdk-web/src/internal/control-client.ts:133]: this seems to be a special header than needs to get passed at the request level

File: packages/client-sdk-web/src/internal/control-client.ts (hunk 2 above)
Comment (cprice404) [reply packages/client-sdk-web/src/internal/control-client.ts:82]: lol

File: packages/client-sdk-web/src/internal/control-client.ts [hunk 4]
```diff
@@ -100,7 +100,7 @@ export class ControlClient<
     return await new Promise<CreateCache.Response>(resolve => {
       this.clientWrapper.createCache(
         request,
-        null,
+        this.authHeaders,
```
Comment (cprice404) [inline packages/client-sdk-web/src/internal/control-client.ts:103]: just curious why you're doing the agent header in an interceptor and the auth header on the request?

File: scripts/build-package.sh [hunk 5]
```diff
@@ -6,7 +6,7 @@ set -e
 echo "building package: ${1}"
 
 pushd ./packages/${1}
-    npm ci
+    npm ci -f
```
Comment (cprice404) [inline scripts/build-package.sh:9]: what does that do?

**Review by cprice404 — COMMENTED** [review]
:nice:

File: scripts/build-package.sh (hunk 5 above)
Comment (bruuuuuuuce) [reply scripts/build-package.sh:9]: it does a force install, I kept getting flaky github errors around `npm ERR! code EEXIST`. We can fix this before we merge back to main, this is just what fixed it and unblocked me for building the 2 sdks on pr

Comment (cprice404) [reply scripts/build-package.sh:9]: okay maybe leave a comment there so it's obvious, then i'm fine with it for now

File: packages/client-sdk-web/src/internal/control-client.ts (hunk 4 above)
Comment (bruuuuuuuce) [reply packages/client-sdk-web/src/internal/control-client.ts:103]: honestly not sure, maybe our interceptors just aren't working correctly? I can change it if you want.

Comment (cprice404) [reply packages/client-sdk-web/src/internal/control-client.ts:103]: it's not urgent but would be nice if they were consistent.  could just file a follow-up ticket.  not a blocker.

**Comment by bruuuuuuuce** [comment]
integration tests should pass once this pr https://github.com/momentohq/control-plane-service/pull/306 is merged and deployed to alpha

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -2,7 +2,7 @@ name: Build
 
 on:
   pull_request:
-    branches: [main]
+    branches: [main, console-mvp]
```
Comment (bruuuuuuuce) [inline .github/workflows/build.yml:5]: I am seeing some strange behavior from earlier versions of nodejs with the web sdk. Removing them for now to unblock us. Will have to come back to this when we merge it into main

**Review by poppoerika — APPROVED** [review]
Looks good to me.

**Pull request merged**